src/main/java/org/folio/circulation/domain/CheckInLogRecord.java

```diff
@@ -0,0 +1,96 @@
+package org.folio.circulation.domain;
+
+import java.util.UUID;
+
+import org.apache.commons.lang3.ObjectUtils;
+import org.joda.time.DateTime;
+
+import io.vertx.core.json.JsonObject;
+
+public class CheckInLogRecord {
+  private static final String ID = "id";
+  private static final String OCCURRED_DATE_TIME = "occurredDateTime";
+  private static final String ITEM_ID = "itemId";
+  private static final String CHECK_IN_SERVICE_POINT_ID = "checkInServicePointId";
+  private static final String PERFORMED_BY_USER_ID = "performedByUserId";
+
+  private final String id;
+  private final DateTime occurredDateTime;
+  private final String itemId;
+  private final String checkInServicePointId;
+  private final String performedByUserId;
+
+  private CheckInLogRecord(Builder builder) {
+    this.id = builder.id;
+    this.occurredDateTime = builder.occurredDateTime;
+    this.itemId = builder.itemId;
+    this.checkInServicePointId = builder.checkInServicePointId;
+    this.performedByUserId = builder.performedByUserId;
+  }
+
+  public static class Builder {
```

Comment:
What is the builder used for? Why are these not constructor parameters instead?

Reply:
Most of the params are strings it will be hard enough to much them, I guess builder will better suite here.

Reply:
Ok, personally, it seems like a lot of extra code for that purpose.